d2/src/test/java/com/linkedin/d2/balancer/strategies/relative/StateUpdaterTest.java

```diff
@@ -163,8 +164,44 @@ public void testInitializePartitionWithSlowStartInitialHealthScore(boolean doNot
 
   }
 
-  @DataProvider(name = "doNotSlowStart")
-  public Object[][] doNotSlowStart()
+  @Test(dataProvider = "trueFalse")
+  public void testInitializePartitionWithDoNotLoadBalance(boolean doNotLoadBalance)
+  {
+    double initialHealthScore = 0.01;
+    D2RelativeStrategyProperties relativeStrategyProperties = new D2RelativeStrategyProperties()
+      .setInitialHealthScore(initialHealthScore);
+    setup(relativeStrategyProperties, new ConcurrentHashMap<>());
+
+    List<TrackerClient> trackerClients = TrackerClientMockHelper.mockTrackerClients(2,
+                                                                                    Arrays.asList(20, 20), Arrays.asList(10, 10), Arrays.asList(200L, 500L), Arrays.asList(100L, 200L), Arrays.asList(0, 0), false, Arrays.asList(false, doNotLoadBalance));
+
+    assertTrue(_stateUpdater.getPointsMap(DEFAULT_PARTITION_ID).isEmpty(), "There should be no state before initialization");
+
+    _stateUpdater.updateState(new HashSet<>(trackerClients), DEFAULT_PARTITION_ID, DEFAULT_CLUSTER_GENERATION_ID, false);
+
+    final TrackerClient trackerClient0 = trackerClients.get(0);
+    final TrackerClient trackerClient1 = trackerClients.get(1);
+
+    assertEquals(_stateUpdater.getPointsMap(DEFAULT_PARTITION_ID).get(trackerClient0.getUri()).intValue(),
+                 (int) (initialHealthScore * RelativeLoadBalancerStrategyFactory.DEFAULT_POINTS_PER_WEIGHT));
+    assertFalse(_stateUpdater.getPartitionState(DEFAULT_PARTITION_ID).getTrackerClientStateMap().get(trackerClient0).isUnhealthy());
```

Comment:
nitpick: this looks like fit into a simple helper method to DRY for line 191-193 and 197-198 to avoid errors.